consensus/core/src/leader_schedule.rs

```diff
@@ -1,36 +1,179 @@
+        self.num_commits_per_schedule
+            .saturating_sub(unscored_committed_subdags_count) as usize
+    }
+
+    pub(crate) fn update_leader_schedule(
+        &self,
+        dag_state: Arc<RwLock<DagState>>,
+        committer: &UniversalCommitter,
+    ) {
+        let _s = self
+            .context
+            .metrics
+            .node_metrics
+            .scope_processing_time
+            .with_label_values(&["LeaderSchedule::update_leader_schedule"])
+            .start_timer();
+
+        let mut dag_state = dag_state.write();
+        let unscored_subdags = dag_state.take_unscored_committed_subdags();
+
+        // TODO: remove this once scoring strategy is finalized
+        let scoring_strategy =
+            if let Ok(scoring_strategy) = std::env::var("CONSENSUS_SCORING_STRATEGY") {
+                tracing::info!(
+                    "Using scoring strategy {scoring_strategy} for ReputationScoreCalculator"
+                );
+
+                let scoring_strategy: Box<dyn ScoringStrategy> = match scoring_strategy.as_str() {
+                    "vote" => Box::new(VoteScoringStrategy {}),
+                    "certified_vote_v1" => Box::new(CertifiedVoteScoringStrategyV1 {}),
+                    "certified_vote_v2" => Box::new(CertifiedVoteScoringStrategyV2 {}),
+                    "certificate" => Box::new(CertificateScoringStrategy {}),
+                    _ => Box::new(VoteScoringStrategy {}),
+                };
+                scoring_strategy
+            } else {
+                tracing::info!(
+                    "Using scoring strategy VoteScoringStrategy for ReputationScoreCalculator"
+                );
+                Box::new(VoteScoringStrategy {})
+            };
+
+        let score_calculation_timer = self
+            .context
+            .metrics
+            .node_metrics
+            .scope_processing_time
+            .with_label_values(&["ReputationScoreCalculator::calculate"])
+            .start_timer();
+        let reputation_scores = ReputationScoreCalculator::new(
+            self.context.clone(),
+            committer,
+            &unscored_subdags,
+            scoring_strategy,
+        )
+        .calculate();
+        drop(score_calculation_timer);
+
+        reputation_scores.update_metrics(self.context.clone());
+
+        self.update_leader_swap_table(LeaderSwapTable::new(
+            self.context.clone(),
+            reputation_scores.clone(),
+            self.context
+                .protocol_config
+                .consensus_bad_nodes_stake_threshold(),
+        ));
+
+        self.context
+            .metrics
+            .node_metrics
+            .num_of_bad_nodes
+            .set(self.leader_swap_table.read().bad_nodes.len() as i64);
+
+        // Buffer score and last commit rounds in dag state to be persisted later
+        dag_state.add_commit_info(
+            reputation_scores.commit_range,
+            reputation_scores.scores_per_authority,
```

Comment:
Just to confirm, the intention now is to write the commit info not on every commit flush and only for the last commit , but only once when the leader schedule gets update (ex every 300 commits), right?